City-of-Helsinki/helsinki-paatokset
# PP-336 & PP-328: Motion attachments and hidden decision attachments

**To test**
- Run `make drush-cim drush-cr`
- If you don't have decisions data yet, run `make ahjo-migrations`
- There should be a new "Non public decision attachments text" config option in the default text settings: https://helsinki-paatokset.docker.so/fi/admin/tools/default-texts (under messages and alerts)
  - Add some text there and make sure you can translate it 

**Motion attachments**
- Run `drush ahjo-proxy:get-motions -v` (import some upcoming meetings if no motions are generated)
- Run `drush ahjo-proxy:check-decision-attachments --motions -v`
  - This should list all motions with attachments. Use the node IDs to open some of them
- The attachments should appear in the sidebar just like on decisions
- If there are no motions with non-public attachments, edit a motion and change the "PublicityClass" property in the JSON data to "Ei Julkinen" for example (doesn't matter, as long as it is not "Julkinen").
- The attachment list should change depending on the following logic:
  - If the link is missing, just the title is printed with an X icon (no empty link)
  - If the attachment is not public, no link is printed and the icon changes to a crossed eye icon. The title is replaced with "Non-public"
  - If there are non-public attachments, the explanation text is printed below the list
- The "Non-public" text should be translatable through UI translation

**Decision attachments**
- To re-download decision attachments, run `drush ahjo-proxy:update-decision-attachments --limit=300 -v` (Adjust the limit parameter or re-run command if you want to go through all decisions)
- Run `drush ahjo-proxy:check-decision-attachments --offset=0 --limit=100 -v` to get debug information about the attachments. You can adjust the offset and limit parameters to go through all decisions you have, if necessary
  - Note: Currently, it looks like all decisions only have one attachment, they are all listed as public and some file URLs are missing (errors / limitations in the API).
- Find a decision with some attachments. Check that it works according to the logic listed above. Edit the JSON data to test the non-public attachment logic
